frontend/src/metabase/visualizations/lib/LineAreaBarRenderer.js

```diff
@@ -966,6 +967,14 @@ export default function lineAreaBar(
 
   parent.render();
 
+  datas.map(data => {
+    data.map(d => {
+      if (d._waterfallValue) {
+        d[1] = d._waterfallValue;
```

Comment:
The series values (positives, negatives) are restored here, after being "smuggled" through `_waterfallValue`. The reason is that up to this point, the series values are still needed for various other computations.

Reply:
It's nice to explicitly track the waterfall value separately from the different bar type series values, but adding attributes to an array always feels messy to me (though, we do it all over). Maybe we could get the same values by taking the positive bar value and subtracting the negative one? Now that I wrote that out it sounds uglier 😄

Reply:
It's actually adding a (semantically hidden) property to an object (not an array) but yeah, I'm not terribly with this either.

I think it's an indication that we shall make (for a near-future follow-up refactoring) a proper structure for the chart series data, vis-a-vis leaving it as a rather loosely-defined object as in the case right now.